doc/api_ref/pubkey.rst

```diff
@@ -120,14 +120,27 @@ Kyber
 
 Post-quantum key encapsulation scheme based on (structured) lattices.
 
+Currently two flavors of Kyber are implemented in separate Botan modules:
+
+ * ``kyber``, that uses Keccak (SHAKE and SHA-3), and that saw some public
+   usage by early adopters.
+ * ``kyber_90s``, that uses AES/SHA-2 instead of Keccak-based primitives.
+   This mode is deprecated and will be removed in a future release.
+
 .. note::
 
-   Currently two modes for Kyber are defined: the round3 specification
-   from the NIST PQC competition, and the "90s mode" (which uses
-   AES/SHA-2 instead of SHA-3 based primitives). The 90s mode Kyber is
-   deprecated and will be removed in a future release.
+   This is a pre-standardized version of the ML-KEM scheme, that saw
+   some public usage by early adopters. The specification is commonly
+   referred to be version 3.01 of the CRYSTALS-Kyber submission to NIST's
+   third round of the PQC competition. It is not compatible with the
+   "Initial Public Draft" version of FIPS 203 for which Botan does not
+   offer an implementation.
+
+ML-KEM
+~~~~~~
 
-   The final NIST specification version of Kyber is not yet implemented.
+Post-quantum key encapsulation scheme based on (structured) lattices,
+as standardized in FIPS 203.
 
```

Comment:
Arbitrary code location:

- credits.rst:166 can be updated with `ML-KEM/Kyber` instead of `Kyber`.
- FFI should not work for ML-KEM yet. But this is future work, I guess.

Reply:
Lets update the credits independently! :)

Re:FFI: The FFI is tricky, actually. `botan_privkey_view_kyber_raw_key` and `botan_pubkey_view_kyber_raw_key` should just work with ML-KEM keys as well, I think.
For `botan_privkey_load_kyber` and `botan_pubkey_load_kyber` the idea of using the buffer length as an indicator for the required parameter set is coming to bite us. The public key lengths are now ambiguous between Kyber and ML-KEM. And the private key of ML-KEM is 64 bytes, regardless of the the parameter set. Probably we should just add new methods `botan_*_mlkem_*()` and deprecate the kyber variants in due time.

Reply:
Re:FFI, see here: https://github.com/randombit/botan/pull/3893/commits/06c817558ce72495225e80aee515c271f716e6c3